src/parsing_routines_matrixnetworks.jl

```diff
@@ -122,3 +122,103 @@ function loadrxnetwork(ft::MatrixNetwork,
 
     ParsedReactionNetwork(rn, nothing)
 end
+
+
+"""
+    For loading networks with complex stoichiomatix ,incidence matrix.
+
+Notes:
+- Assumed that column of complex stoichiomatrix represents composition of reaction complexes,
+  with positive entries of size num_of_species x num_of_complexes, where
+  the non-zero positive entries in the kth column denote stoichiometric
+  coefficients of the species participating in the kth reaction complex.
+
+- The complex incidence matrix, is number of complexes by number of reactions with
+  Bᵢⱼ = -1, if the i'th complex is the substrate of the j'th reaction,
+         1, if the i'th complex is the product of the j'th reaction,
+         0, otherwise
+"""
+#  for dense matrices
+function LoadReacCompNetwork(rateexprs::AbstractVector,
```

Comment:
I think it would make more sense to stick with the current api. So perhaps make a new structure to hold all this input, and then use dispatch to call a version of `loadrxnetwork` that depends on it. That function should then return a `ParsedReactionNetwork` as the other versions do. i.e.
```julia
struct ComplexMatrixNetwork{S,T,U,V,W} where {S <: AbstractVector, T <: AbstractMatrix, U <: AbstractMatrx, V <: AbstractVector, W <: AbstractVector}
    rateexprs::S
    complexstoichmat::T
    incidencemat::U
    species::V
    params::W
end
ComplexMatrixNetwork(rateexprs, complexstoichmat, incidencemat) = ComplexMatrixNetwork(rateexprs, complexstoichmat, incidencemat, Any[], Any[])

function loadrxnetwork(cmnetwork::T) where {T <: ComplexMatrixNetwork}
    ...
    return ParsedReactionNetwork(rs, nothing)
end
```

Reply:
little offtrack query ,sorry if it sounds silly!
@isaacsas , should we restrict `complexstoichmat` ,and `incidencemat` to `Matrix` in one method and `SparseMatrixCSC` in other ? 
(Instead of `AbstractMatrix` and `SparseMatrixCSC` as used in https://github.com/SciML/ReactionNetworkImporters.jl/blob/master/src/parsing_routines_matrixnetworks.jl)

The reason being, when we use `AbstractMatrix`, it sort of takes `SparseMatrixCSC` also into consideration. Does this not risk, using a dense version of code running with `sparse` inputs? 
```julia
function foo(A::AbstractMatrix)
    a::AbstractMatrix = 2*A # some operation
    return a
end
julia> mymat=sprand(Float64,2,2,0.6)
julia> mymat=sparse([2 0;0 2])
2×2 SparseMatrixCSC{Int64, Int64} with 2 stored entries:
 2  ⋅
 ⋅  2

julia> foo(mymat)
2×2 SparseMatrixCSC{Int64, Int64} with 2 stored entries:
 4  ⋅
 ⋅  4
```

Reply:
> little offtrack query ,sorry if it sounds silly!
> @isaacsas , should we restrict `complexstoichmat` ,and `incidencemat` to `Matrix` in one method and `SparseMatrixCSC` in other ?
> (Instead of `AbstractMatrix` and `SparseMatrixCSC` as used in https://github.com/SciML/ReactionNetworkImporters.jl/blob/master/src/parsing_routines_matrixnetworks.jl)

Sorry again !, I understood this after experimenting a little on my own why `AbstractMatrix` and `SparseMatrixCSC` is used

Reply:
> I think it would make more sense to stick with the current api. So perhaps make a new structure to hold all this input, and then use dispatch to call a version of `loadrxnetwork` that depends on it. That function should then return a `ParsedReactionNetwork` as the other versions do. i.e.
> 
> ```julia
> struct ComplexMatrixNetwork{S,T,U,V,W} where {S <: AbstractVector, T <: AbstractMatrix, U <: AbstractMatrx, V <: AbstractVector, W <: AbstractVector}
>     rateexprs::S
>     complexstoichmat::T
>     incidencemat::U
>     species::V
>     params::W
> end
> ComplexMatrixNetwork(rateexprs, complexstoichmat, incidencemat) = ComplexMatrixNetwork(rateexprs, complexstoichmat, incidencemat, Any[], Any[])
> 
> function loadrxnetwork(cmnetwork::T) where {T <: ComplexMatrixNetwork}
>     ...
>     return ParsedReactionNetwork(rs, nothing)
> end
> ```

Hey @isaacsas , do you think we should make this `mutable struct` ?
Reason being, `species` could be sent `Any[]` by the user, and then we create 
`    isempty(species) && (species = [funcsym(:S,t,i) for i = 1:numspecs])
`

Reply:
No because that species vector is for the new ReactionSystem. We don’t need to change the input to the function at all.

Reply:
i added two structs. `ComplexMatrixNetwork` and `ComplexSparseMatrixNetwork` to achieve this